src/main/java/org/apache/mesos/hdfs/state/LiveState.java

```diff
@@ -1,103 +1,123 @@
 package org.apache.mesos.hdfs.state;
 
+import com.google.common.base.Predicate;
+import com.google.common.collect.Sets;
 import com.google.inject.Singleton;
-import org.apache.commons.logging.Log;
-import org.apache.commons.logging.LogFactory;
 import org.apache.mesos.Protos;
 import org.apache.mesos.hdfs.util.HDFSConstants;
 
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
 
 @Singleton
 public class LiveState {
-  private static final Log log = LogFactory.getLog(LiveState.class);
-  private final Map<Protos.TaskID, String> taskHostMap;
-  private final Set<Protos.TaskID> journalNodes;
-  private final Set<Protos.TaskID> nameNodes;
-  private final Map<Protos.TaskID, String> taskSlaveMap;
-  private final Set<String> journalNodeHosts;
-  private final Set<String> nameNodeHosts;
-  private final HashSet<Protos.TaskID> stagingTasks;
+  private Set<Protos.TaskInfo> stagingTasks = new HashSet<>();
+  private AcquisitionPhase currentAcquisitionPhase = AcquisitionPhase.JOURNAL_NODES;
+  private LinkedHashMap<Protos.TaskID, Protos.TaskStatus> runningTasks = new LinkedHashMap<>();
 
-  public LiveState() {
-    taskHostMap = new HashMap<>();
-    taskSlaveMap = new HashMap<>();
-    journalNodes = new HashSet<>();
-    nameNodes = new HashSet<>();
-    journalNodeHosts = new HashSet<>();
-    nameNodeHosts = new HashSet<>();
-    stagingTasks = new HashSet<>();
+  public void addStagingTask(Protos.TaskInfo taskInfo) {
+    stagingTasks.add(taskInfo);
   }
-  
-  public Map<Protos.TaskID, String> getTaskHostMap() {
-    return taskHostMap;
+
+  public int getStagingTasksSize() {
+    return stagingTasks.size();
+  }
+
+  public void removeStagingTask(final Protos.TaskID taskID) {
+    Set<Protos.TaskInfo> toRemove = new HashSet<>();
+    for (Protos.TaskInfo taskInfo : stagingTasks ) {
```

Comment:
Why store TaskInfos in the HashSet if you're only counting and later removing by taskID? If you don't need the rest of the TaskInfo, you could just make it a set of TaskIDs, and use HashSet.remove(taskID) instead of having to iterate. There can only be one stagingTask with that ID anyway.